quint/src/runtime/compile.ts

```diff
@@ -59,46 +66,92 @@ function errorContext(errors: ErrorMessage[]): CompilationContext {
   }
 }
 
+/**
+ * Extract a compiled value of a specific kind via the module name and kind.
+ *
+ * @param ctx compilation context
+ * @param moduleName module name to lookup at
+ * @param defName definition name
+ * @param kind definition kind
+ * @returns the associated compiled value, if it uniquely defined, or undefined
+ */
+export function
+  contextLookup(ctx: CompilationContext,
+                moduleName: string,
+                defName: string,
+                kind: ComputableKind): Either<string, Computable> {
+  const moduleTable = ctx.lookupTable.get(moduleName)
+  if (!moduleTable) {
+    return left(`Module ${moduleName} not found`)
+  }
+  const defs = moduleTable.valueDefinitions.get(defName)
+  if (!defs) {
+    return left(`Definition ${moduleName}::${defName} not found`)
+  }
+  if (defs.length !== 1) {
+    return left(`Multiple definitions (${defs.length}) of ${moduleName}::${defName} found`)
+  }
```

Comment:
Should this be using `lookupValue`? Although there's no real encapsulation, we should only manipulate `lookupTable` with the accompanying functions.

Reply:
I have been thinking about that. The issue is that we do not have a scope id in this case. So I will keep it as it is atm. But it would be good to come back to the lookup API at some point.